pkg/common/cns-lib/volume/manager.go

```diff
@@ -763,3 +765,57 @@ func (m *defaultManager) RelocateVolume(ctx context.Context, relocateSpecList ..
 	}
 	return res, err
 }
+
+// ConfigureVolumeACLs configures ACLs on the CNS volume
+func (m *defaultManager) ConfigureVolumeACLs(ctx context.Context, spec cnstypes.CnsVolumeACLConfigureSpec) error {
+	log := logger.GetLogger(ctx)
+	err := validateManager(ctx, m)
+	if err != nil {
+		return err
+	}
+	// Set up the VC connection
+	err = m.virtualCenter.ConnectCns(ctx)
+	if err != nil {
+		log.Errorf("ConnectCns failed with err: %+v", err)
+		return err
+	}
+
+	var task *object.Task
+	var taskInfo *vim25types.TaskInfo
+
+	task, err = m.virtualCenter.CnsClient.ConfigureVolumeACLs(ctx, spec)
+	if err != nil {
+		log.Errorf("CNS ConfigureVolumeACLs failed from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return err
+	}
+
+	// Get the taskInfo
+	taskInfo, err = cns.GetTaskInfo(ctx, task)
+	if err != nil || taskInfo == nil {
+		log.Errorf("failed to get taskInfo for ConfigureVolumeACLs task from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
+		return err
```

Comment:
if the control flow enters this path when `taskInfo == nil`, then we're returning `nil` and the caller might assume that `ConfigureVolumeACLs` succeeded?

Reply:
Looks like we are following the same for all APIs. taskInfo will never be nil, right ? If wait for vc task fails, then we get back err object. Let me know if we are good to remove `taskInfo == nil` check